src/Scripting/IBindings.h

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include <sol/sol.hpp>
+
+class IBindings {
+ public:
+    virtual ~IBindings() = default;
+    virtual sol::table getBindingTable() = 0;
```

Comment:
This interface combines two functions:
1. Binding factory.
2. Cache for said binding factory.

I suggest we make this a single responsibility class, so it'll be `createBindingTable`, and caching (if any) should be done at the caller's side. Also `createBindingTable` will then be `const`.

Reply:
Thinking about it, `createBindingTable` should take `sol::state_view` as an arg, and then we'll have one less implicit arg there.

Reply:
Done. Renamed to `createBindingTable`, transformed into a const method, and `sol::state_view` is passed as arg.
This is removing the need of keeping the solState as a member on all the Bindings classes.
This is showing me another little issue with how the `GameLuaBindings` class is taking care of the `InfoQueryTable` . Something that should be its responsibility. For now I'm doing a little hack by transforming the instance to a static member but I'm planning to move it elsewhere instead.